src/main/java/calculator/Calculator.java

```diff
@@ -0,0 +1,139 @@
+        if (token.equals("(")) return 1;
+        if (token.equals("+") || token.equals("-")) return 2;
+        if (token.equals("*") || token.equals("/")) return 3;
+        return 4;
+    }
+
+    private List<String> parse(String infix) {
+        List<String> postfix = new ArrayList<>();
+        Deque<String> stack = new ArrayDeque<>();
+        StringTokenizer tokenizer = new StringTokenizer(infix, delimiters, true);
+        String prev = "";
+        String curr;
+        while (tokenizer.hasMoreTokens()) {
+            curr = tokenizer.nextToken();
+            if (!tokenizer.hasMoreTokens() && isOperator(curr)) {
+                flag = false;
+                return postfix;
+            }
+            if (curr.equals(" ")) continue;
+            if (isFunction(curr)) stack.push(curr);
+            else if (isDelimiter(curr)) {
+                if (curr.equals("(")) stack.push(curr);
+                else if (curr.equals(")")) {
+                    while (!"(".equals(stack.peek())) {
+                        postfix.add(stack.pop());
+                        if (stack.isEmpty()) {
+                            flag = false;
+                            return postfix;
+                        }
+                    }
+                    stack.pop();
+                    if (!stack.isEmpty() && isFunction(stack.peek())) {
+                        postfix.add(stack.pop());
+                    }
+                } else {
+                    if (curr.equals("-") && (prev.equals("") || (isDelimiter(prev) && !prev.equals(")")))) {
+                        curr = "u-";
+                    } else {
+                        while (!stack.isEmpty() && (priority(curr) <= priority(stack.peek()))) {
+                            postfix.add(stack.pop());
+                        }
+                    }
+                    stack.push(curr);
+                }
+
+            } else {
+                postfix.add(curr);
+            }
+            prev = curr;
+        }
+
+        while (!stack.isEmpty()) {
+            if (isOperator(stack.peek())) postfix.add(stack.pop());
+            else {
+                flag = false;
+                return postfix;
+            }
+        }
+        return postfix;
+    }
+
+
+        private static int calc(List<String> postfix) {
+            Deque<Integer> stack = new ArrayDeque<>();
+            for (String x : postfix) {
+                switch (x) {
+                    case "+":
+                        stack.push(stack.pop() + stack.pop());
+                        break;
+                    case "-": {
+                        int b = stack.pop(), a = stack.pop();
+                        stack.push(a - b);
+                        break;
+                    }
+                    case "*":
+                        stack.push(stack.pop() * stack.pop());
+                        break;
+                    case "/": {
+                        int b = stack.pop(), a = stack.pop();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use one line for each declaration, it enhances code readability.](https://app.codacy.com/app/thejerome/IFMO_Servlets_20190915/pullRequest?prid=4510206)